Updated the extensions section (#5613)
jenkins-infra/jenkins.io: content/doc/developer/architecture/extensions.adoc

title: Extensions
layout: developersection
wip: true
---

> Jenkins defines extension points, which are interfaces or abstract classes that model an aspect of its behavior. Those interfaces define contracts of what need to be implemented, and Jenkins allows plugins to contribute those implementations. In general, all you need to do to register an implementation is to mark it with `@Extension`.

Reviewer: ```suggestion
Jenkins defines extension points, which are interfaces or abstract classes that model an aspect of its behavior.
The interface or class defines contracts of what needs implementation, and Jenkins allows plugins to contribute those implementations.
To register an implementation, make sure you mark it with `@Extension`.
```
A small suggestion to update the content to be one sentence per line (making it easier to read and review), and adjust some of the wording to flow a bit better.

> https://www.jenkins.io/doc/developer/extensions/[The extensions index]  is a useful list of the extension points used in Jenkins and Jenkins plugins.

Reviewer: ```suggestion
link:https://www.jenkins.io/doc/developer/extensions/[The extensions index] is a useful list of the extension points used in Jenkins and Jenkins plugins.
```
adding in the `link:` macro to make sure the link is directed properly and a small spacing update

> More information can be found in the https://www.jenkins.io/doc/developer/extensibility/[Extensibility section].

Reviewer: ```suggestion
More information can be found in the link:https://www.jenkins.io/doc/developer/extensibility/[Extensibility section].
```
again, just adding the link macro here

title: Extensions
layout: developersection
wip: true
---

> Jenkins defines extension points, which are interfaces or abstract classes that model an aspect of its behavior. The interface or class defines contracts of what needs implementation, and Jenkins allows plugins to contribute those implementations.

Reviewer: ```suggestion
Jenkins defines extension points, which are interfaces or abstract classes that model an aspect of its behavior.
The interface or class defines contracts of what needs implementation, and Jenkins allows plugins to contribute those implementations.
```
I saw that the earlier modification was to start another run (since the previous one had failed), but would it be possible to update this back to be two separate lines?  Thanks!
